tikv/tikv
# log-backup: adapt user-defined TLS config

Signed-off-by: Yu Juncen <yujuncen@pingcap.com>



### What is changed and how it works?

Issue Number: Close #12668

What's Changed:
- Added a new metastore type `LazyEtcd`, which supports connecting with TLS.
- Modify the `Endpoint::new`, now it receives `MetaStore` instead of `endpoints`.


```commit-message
log-backup: adapt user-defined TLS config
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
I have tested it by:
1. Create a cluster by `tiup`.
2. Enable TLS by `tiup cluster tls <cluster-name> enable`.
3. Run log backup, there should not be errors or panic.
- No code


### Release note 

```release-note
Fix a bug that caused PiTR cannot work if some of `security.{ca-path, cert-path, key-path}` configured.
```
